### DISCUSSION TITLE
Misskeyのフロントエンドが重い問題についての情報を集めるためのスレッド (What are the reasons why Misskey frontend seems heavy)

### DISCUSSION THREAD

**Discussion by Sayamame-beans** [misskey-dev/misskey#15145] [General]
Misskeyのフロントエンドが重いとはよく言われますが、実際どの部分が重いのだろうかという問題があると思うので、その情報やissueを集めるために開きました。
(公式アプリとかもありませんし、)バッテリー持ち的な意味でもUX的な意味でも出来れば軽量な方が良いと考えられます。

**Comment by syuilo** [comment]
2024.10.1以降は大幅に改善した認識だわね

**Comment by SINtyanneru** [comment]
(私の環境だけかもだけど、長期間タブを開きぱなしにしてたら挙動が重くなったことがよくあったり)

**Comment by Sayamame-beans** [comment]
とりあえずTL遡ってると重くなるというのはよく遭遇しますね…(多分全部残っているため)

**Comment by samunohito** [comment]
①TLを遡行すると重い

遡行すれば遡行するほどMkNoteが増えていく状態になっています。
一定数のノートを読み込んだら配列をトリミングするなどの処理が必要だと考えます

**Comment by syuilo** [comment]
遡ると遅いのは2024.10.1以降でほぼ解決したはず

**Reply by Sayamame-beans** [reply]
まだ重いような気がします…?(スマホ上でPWAを使用しています)
遡った状態で何もしていないタイミングは何となく重くないようになってる感じがしますが、スクロールしたりリアクションしようとしたりするとまだ重さを感じるような印象でしょうか…?

**Reply by syuilo** [reply]
お使いのデバイスの種類って何かしら

**Reply by Sayamame-beans** [reply]
種類… Android(v12)ですね
手持ちのiPhoneでも試したところ、描画が遅れる(表示されるまで単色)だけでスクロールについての重さはあまり気にならなそうかもしれないです…? (リアクションは重い)

**Reply by syuilo** [reply]
Androidのブラウザのバージョンが古いと軽量化の効果が期待できないかもしれないわね

**Reply by Sayamame-beans** [reply]
`Chrome 131.0.6778.105`でしたね…

**Reply by kakkokari-gtyih** [reply]
Androidはnote render skippingを有効にしていると、かえってスクロールがカクつくという報告を複数見かけている

**Reply by syuilo** [reply]
手元にAndroidがないから検証が困難だわね

**Comment by kekyo** [comment]
全く参考になるかわからないですが（おひとり様サーバー立てて1日目）、ブラウザ側でublock originとABPが有効になってると猛烈に遅くなりました。
サイト除外設定でオフにする事で、重さを感じなくなったので、何か特定のエレメントを再帰的に探索して大量に存在するから遅くなるとか、そういうことかも知れないと想像しました。

**Reply by kekyo** [reply]
環境書くのを忘れていました。私もFirefoxで起きています。Ubuntu 22.04/24.04で配信中の最新、Windows 11で配信中の最新です。

**Comment by tassoman** [comment]
Sometimes happens loading is delayed, like 100% cpu is running. Windows 10 Firefox.
Only happens on first load. I mean, background color is here, but empty page for about 3-5 seconds.

**Reply by samunohito** [reply]
私もFirefoxユーザーです。同じ現象に悩まされました。

https://github.com/misskey-dev/misskey/issues/14364

この問題が関連している可能性があります。
実際、tabler-iconsを取り除いているサーバでは今のところ現象が発生していません。

**Reply by tassoman** [reply]
I think so! The on loading issue, looks like related to that.
Another different thing pulling my CPU are horizontal tickers. If I set just one ticker, my CPU goes 22% (intel I7 4th gen).
Two tickers are 24-30% (RSS + federated servers).
